binary_to_decimal/binary_to_decimal.js

```diff
@@ -0,0 +1,53 @@
+// ----------------------------------
+// Function binaryToDecimal
+
+function binaryToDecimal(binary_number){
+    const length = binary_number.toString().length; 
+    const string = binary_number.toString();
+    let binaryVal = true; //default value
+
+    for(let i = 0; i < length; i++){  //binary check - compare string
+        if(string[i] == "0" || string[i] == "1"){}
+        
+        else {
+            binaryVal = false;            
+        }
+    }
+
+    if(binaryVal == false){
+        alert("Wrong input number, the program needs to be restarted.");
+    }
+
+// ----------------------------------
+// Conversion to decimal number
+
+    if(binaryVal){
```

Comment:
Tahle podmínka je zbytečná, když na řádku 18 použiješ `return`, ten přeruší funkci a sem už kód nedojde, neboli máš garanci, že pokud sem dojde, binaryVal bude `true`. Kód bude čistší :)

```javascript
return alert("Wrong input number, the program needs to be restarted.");
```

Reply:
Nevedel som, že v for loope sa dá použiť _return_, funguje to v tomto prípade rovnako ako _break_?

Reply:
`break` ti vyleze pouze z bloku cyklu, `return` ti vyleze z celé funkce